W-20130245-troubleshoot-agent-networks-ie (#570)
mulesoft/docs-code-builder: modules/ROOT/pages/af-troubleshoot-agent-networks.adoc

= Troubleshoot Agent Networks
:page-aliases: af-troubleshoot-anypoint-code-builder.adoc

Resolve common agent network errors and connectivity issues by enabling verbose logging, message logging policies, and distributed tracing.

[NOTE]

> You can use Agent Visualizer to xref:agent-visualizer::troubleshoot-agent-network.adoc[view metrics, logs, and traces] for your agents and MCP servers in your network

Reviewer: ```suggestion
Use Agent Visualizer to xref:agent-visualizer::troubleshoot-agent-network.adoc[view metrics, logs, and traces] for your agents and MCP servers in your network
```

== Enable Logging to Debug Agent Brokers

> You can enable logging and log levels from CloudHub 2.0 or Runtime Fabric.

Reviewer: ```suggestion
Enable logging and log levels from CloudHub 2.0 or Runtime Fabric:
```

> 1. In Runtime Manager, navigate to the application for the agent broker.

Reviewer: Do numbers render correctly for ascii docs?

2. Choose one of these options to configure logging:
+
* xref:cloudhub-2::ch2-manage-app-logs.adoc#configure-log-levels-and-categories[CloudHub 2.0]
* xref:runtime-fabric::deploy-to-runtime-fabric.adoc#configure-logging-and-log-levels[Runtime Fabric]
3. Set the log level to Debug for these classes:
+
* `com.mulesoft.modules.agent.broker`
* `org.mule.service.http.impl.service.HttpMessageLogger`

[IMPORTANT]
HTTP wire logging displays HTTP headers and payload in plain text. Avoid leaving it enabled indefinitely or unsupervised, as it can negatively impact API performance.

[[enable-message-logging]]
== Enable Message Logging Policy

> Use API Manager or manually edit to the agent network file to add a message logging policy. 

Reviewer: ```suggestion
Use API Manager or manually edit the agent network file to add a message logging policy. 
```

=== Using API Manager

> . In API Manage, select the API, agent, or MCP server for which you want to apply the policy. 

Reviewer: ```suggestion
. In API Manager, select the API, agent, or MCP server instance for which you want to apply the policy. 
```

. Follow the steps in xref:gateway::flex-managed-message-log.adoc#configure-message-logging-in-connected-mode[Configure Message Logging in Connected Mode] and add these settings:
+
* In the Messages field, enter `#[attributes]`.
* Select the checkboxes for **Before Calling API** and **After Calling API**.
. Click **Save**.

Afterward, you should see a Message Log option for the API, agent, or MCP server.

[NOTE]
If you redeploy the network, this policy is deleted.

=== Using the Agent Network File

> In the `connection` section of the API, agent, or MCP server you want to log, add the following inside `policies`:

Reviewer: ```suggestion
In the `connection` section of the API, agent, or MCP server instance you want to log, add the following inside `policies`:
```
